src/arch/cuda/hybridAPI/hapi_impl.cpp

```diff
@@ -329,58 +319,78 @@ static void hapiMapping(char** argv) {
   CmiUnlock(csv_gpu_manager.device_mapping_lock);
 #endif
 
-  // Process device communication buffer parameters (in MB)
-  int input_comm_buffer_size;
-  if (CmiGetArgIntDesc(argv, "+gpucommbuffer", &input_comm_buffer_size,
-        "GPU communication buffer size (in MB)")) {
-    if (CmiMyRank() == 0) {
-      // Round up size to the closest power of 2
-      size_t comm_buffer_size = (size_t)input_comm_buffer_size * 1024 * 1024;
-      int size_log2 = std::ceil(std::log2((double)comm_buffer_size));
-      csv_gpu_manager.comm_buffer_size = (size_t)std::pow(2, size_log2);
+  // Check if user opted out of shared memory optimizations for
+  // inter-process GPU messaging
+  bool no_shm = false;
+  if (CmiGetArgFlagDesc(argv, "+gpunoshm",
+        "disable shared memory optimizations for inter-process GPU messaging")) {
+    no_shm = true;
+    if (CmiMyPe() == 0) {
+      CmiPrintf("HAPI> Disabled shared memory optimizations for inter-process GPU messaging\n");
     }
   }
 
-  if (CmiMyPe() == 0) {
-    CmiPrintf("HAPI> GPU communication buffer size: %zu MB "
-        "(rounded up to the nearest power of two)\n",
-        csv_gpu_manager.comm_buffer_size / (1024 * 1024));
+  if (CmiMyRank() == 0) {
+    if (no_shm) csv_gpu_manager.use_shm = false;
   }
 
-  CmiNodeBarrier(); // Ensure device communication buffer size is set
+  CmiNodeBarrier();
 
-  // Create device communication buffers
-  // Should only be done by device representative threads
-  if (cpv_device_rep) {
-    DeviceManager* dm = csv_gpu_manager.device_map[CmiMyPe()];
+  if (csv_gpu_manager.use_shm) {
```

Comment:
Device communication buffers and CUDA IPC event pool are only created when the user does not opt out of shared memory optimizations.